padre_meddea/io/amptek.py

```diff
@@ -0,0 +1,159 @@
+        in_data_section = False
+        in_calib_section = False
+        in_roi_section = False
+        in_dp5config_section = False
+        in_status_section = False
+        in_spectrum_meta_section = False
+        calib_data = {}
+        roi_data = {}
+        data = []
+        line_number = 0
+        hdr = Header()
+        meta = {}
+        for line in fp:
+            #  figure out what section of the file we are in
+            # the order used here is the order expected in the file
+            # once in new section you are done with the previous section
+            if line.count("PMCA SPECTRUM"):
+                in_spectrum_meta_section = True
+                continue
+            if line.count("CALIBRATION"):
+                in_calib_section = True
+                in_spectrum_meta_section = False
+                continue
+            if line.count("ROI"):
+                in_roi_section = True
+                in_calib_section = False
+                in_spectrum_meta_section = False
+                continue
+            if line.count("DATA"):
+                in_data_section = True
+                in_roi_section = False
+                continue
+            if line.count("END"):
+                in_data_section = False
+            if line.count("DP5 CONFIGURATION"):
+                in_dp5config_section = True
+                in_data_section = False
+                continue
+            if line.count("DPP STATUS"):
+                in_dp5config_section = False
+                in_status_section = True
+                continue
+
+            # now gather information
+            if in_spectrum_meta_section:  # this is a measurement metadata tag
+                label = line.split("-")[0].strip()
+                value = line.split("-")[1].strip()
+                try:
+                    value = float(value)
+                except ValueError:
+                    pass
+                hdr.append((label, value))
+            if in_calib_section:
+                if line.count("-"):
+                    continue
+                else:
+                    channel = float(line.split(" ")[0])
+                    energy = float(line.split(" ")[1])
+                    calib_data.update({channel: energy})
+            if in_roi_section:
+                roi_data.update({int(line.split(" ")[0]): int(line.split(" ")[1])})
+            if in_data_section and not (line.count("<<") == 1):
+                data.append(int(line))
+            if in_dp5config_section and not (line.count("<<") == 1):
+                keyword = line.split("=")[0].upper()
+                value = line.split("=")[1].split(";")[0].strip()
+                description = line.split("=")[1].split(";")[1].strip()
+                try:
+                    value = float(value)
+                except ValueError:
+                    pass
+                hdr.append((keyword.replace(" ", "")[0:7], value, description))
```

Comment:
Running Line 111 returns the following warnings: 

WARNING: VerifyWarning: Keyword name 'DESCRIPTION' is greater than 8 characters or contains characters not allowed by the FITS standard; a HIERARCH card will be created. [astropy.io.fits.card]
WARNING: VerifyWarning: Keyword name 'THRESHOLD' is greater than 8 characters or contains characters not allowed by the FITS standard; a HIERARCH card will be created. [astropy.io.fits.card]
WARNING: VerifyWarning: Keyword name 'LIVE_MODE' is greater than 8 characters or contains characters not allowed by the FITS standard; a HIERARCH card will be created. [astropy.io.fits.card]
WARNING: VerifyWarning: Keyword name 'PRESET_TIME' is greater than 8 characters or contains characters not allowed by the FITS standard; a HIERARCH card will be created. [astropy.io.fits.card]
WARNING: VerifyWarning: Keyword name 'LIVE_TIME' is greater than 8 characters or contains characters not allowed by the FITS standard; a HIERARCH card will be created. [astropy.io.fits.card]
WARNING: VerifyWarning: Keyword name 'REAL_TIME' is greater than 8 characters or contains characters not allowed by the FITS standard; a HIERARCH card will be created. [astropy.io.fits.card]
WARNING: VerifyWarning: Keyword name 'START_TIME' is greater than 8 characters or contains characters not allowed by the FITS standard; a HIERARCH card will be created. [astropy.io.fits.card]
WARNING: VerifyWarning: Keyword name 'SERIAL_NUMBER' is greater than 8 characters or contains characters not allowed by the FITS standard; a HIERARCH card will be created. [astropy.io.fits.card]
ERROR: ValueError: FITS header values must contain standard printable ASCII characters; '27°C' contains characters not representable in ASCII or non-printable characters. [astropy.io.fits.card]

Reply:
All of those warnings can be ignored. The ERROR at the bottom is what is important but you already referred to it in your previous comment.